dronecan/protocol/7.NodeCapabilities.uavcan

```diff
@@ -0,0 +1,59 @@
+# NodeCapabilities messages
+
+# this message contains information about the capabilities of the node sending the messgage,
+# or if the authoritative flag is set, then it announces the mininum capabilities of all nodes on the bus
+
+# a node should not activate a capability until it has seen an
+# authoritative announcement that the mininum capability on the bus
+# includes the corresponding capability. For example, if a node is
+# capable of doing 8MBit FDCAN then it should set the
+# FLAG_FDCAN_4M_SUPPORT in its announcements, but not start using 4M
+# FDCAN until it has seen an authoritative message with the
+# FLAG_FDCAN_4M_SUPPORT bit set in flags.
+
+# on receipt of this message all nodes should set their active
+# capabilities to the minimum of the received message capabilities and
+# their current capabilities. For example, this means if a node is
+# currently using FDCAN 4M and it sees a packet (whether authoritative
+# or not) without the FDCAN 4M flag set then it should stop using
+# FDCAN and fall back to the next lowest level (in the case of 4M
+# FDCAN this would be 1M bxCAN)
+
+# the central server (usually the DNA server) collates the
+# capabilities of all nodes on the bus and is responsible for
+# announcing the minimum capabilities with the authoritative flag
+# set. The central server will also send an announcement with its own
+# capabilities, which may be higher than the minimum on the bus. This
+# announcement of its own capabilities should not have the
+# authoritative flag set.
+
+# the central server may have a (possibibly configurable) delay
+# between when an increase in minimum network capability happens and
+# when it announces the upgraded capability. A delay on startup is a
+# good idea to allow time for legacy nodes to startup
+
+uint32 FLAG_AUTHORITATIVE         = 1
+# the PUBLISHER_ONLY bit should be set if this node only publishes data, most sensor
+# nodes should set this bit
+uint32 FLAG_PUBLISHER_ONLY        = 2
+uint32 FLAG_FDCAN_4M_SUPPORT      = 4
```

Comment:
Missing BRS support. CAN-FD can be used without setting the BRS or having a different data rate.

Reply:
that is a question for @bugobliterator 
note that we don't really want to support all possible combinations

Reply:
Wouldn't lack of BRS support be implicit in the bitrate flags with the supported bitrate being limited to 1M (FLAG_FDCAN_1M_SUPPORT)?

Reply:
After some discussions on Discord (my kids are laughing at me for using Discord BTW. Thanks for that) I think the best design here is to define a set of abstract, Level-1 standards and given them names. For example:

1. DRONECAN_COMPAT
2. DRONECAN_RELIABLE
3. DRONECAN_FAST
4. DRONECAN_FAST_SI

where each is defined with a separate specification for your L1 bus. This would include all of the considerations like arbitration rate, data rate, sample point, etc. My example here would optimize these settings for:

1. compatibility with existing systems
2. using FD (with the improved checksum) at a slow data rate and without BRS to minimize SI issues.
3. using FD at a fast datarate which you fix.
4. using FD with Signal Improvement transceivers to achieve 8Mb data rates.

That said, it may introduce some complexity into firmwares to require runtime reconfiguration to a minimum supported standard. You might allow this message to simply report to the user that "node x is not compatible with this system" and provide a link to the manufacturer's reported help URL. That URL can then instruct the user on how to reconfigure the LRU into one of the four modes. This suggests you need a "can dynamically reconfigure" flag in the message.